### PR TITLE
Migration bug fix: Strip leading zeroes in docketNumber

### PR DESCRIPTION
**Pull request by ericsorenson**

### CODE REVIEW

**Review by kkoskelin — APPROVED** [review]

**Review by rachaelparris — APPROVED** [review]

File: shared/src/business/useCases/getCaseInteractor.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ exports.getCaseInteractor = async ({ applicationContext, caseId }) => {
         applicationContext,
         caseId,
       });
-  } else if (Case.isValidDocketNumber(caseId)) {
+  } else {
```
Comment (kkoskelin) [inline shared/src/business/useCases/getCaseInteractor.js:64]: Same if/else within `getCaseForPublicDocketSearchInteractor` and `getPublicCaseInteractor`

Comment (kkoskelin) [reply shared/src/business/useCases/getCaseInteractor.js:64]: `Case.isValidDocketNumber` is a disease that must be eradicated.

**Review by kkoskelin — CHANGES_REQUESTED** [review]
I think we need to remove that `isValidDocketNumber` function along with all occurrences which call it.

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**